projects/ngx-formentry/src/form-entry/helpers/js-expression-helper.ts

```diff
@@ -164,6 +166,68 @@ export class JsExpressionHelper {
     return height && weight && refSectionObject ? formattedSDValue : null;
   }
 
+  getAgePosition(age) {
+    if (age >= 70) {
+      return 0;
+    } else if (age >= 65) {
+      return 1;
+    } else if (age >= 60) {
+      return 2;
+    } else if (age >= 55) {
+      return 3;
+    } else if (age >= 50) {
+      return 4;
+    } else if (age >= 45) {
+      return 5;
+    }
+    return 6;
+  }
+
+  getSBPPosition(sbp) {
+    if (sbp >= 180) {
+      return 0;
+    } else if (sbp >= 160) {
+      return 1;
+    } else if (sbp >= 140) {
+      return 2;
+    } else if (sbp >= 120) {
+      return 3;
+    } else if (sbp >= 50) {
+      return 4;
+    } else if (sbp >= 45) {
+      return 5;
+    }
+    return 6;
+  }
+
+  getBMIPosition(bmi) {
+    if (bmi >= 35) {
+      return 4;
+    } else if (bmi >= 30) {
+      return 3;
+    } else if (bmi >= 25) {
+      return 2;
+    } else if (bmi >= 20) {
+      return 1;
+    }
+    return 0;
+  }
+
+  // Order Sex -> Age -> SBP -> Smoker -> BMI
+  calculateBMI(height, weight) {
+    return weight / (height * height);
+  }
+
+  calcNonLabCVDRisk({ age, sex, smoker, height, weight, sbp }) {
```

Comment:
Shouldn't we just have the BMI as a parameter? Even if we use the height / weight on the form, we can separately calculate the BMI without needing to do it here.

Reply:
I don't know about that, because looking at the forms, we mostly have `height` and `weight` and they use those to calculate the `BMI` , are you saying we can have the form pass the `BMI` value? If it can then that's better.

Reply:
Yeah... I mean it's just a matter of adding a BMI field to the form, right? And I always think it's best for these things to be user-visible.